<a href="https://colab.research.google.com/github/petrus1210/pemograman-dasar/blob/main/ATS_PRAKTIKUM_KELOMPOK_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
Nama Kelompok 1

-Rifa Halwani (4232211015)
-Petrus Randanu (4232411007)
-Adi Candra Edita (4232411022)

In [1]:
#SOAL NO 2
def hitung_efisiensi(input_energi, output_energi):
    """
    Fungsi ini menghitung efisiensi turbin.
    Parameter:
        input_energi (float): Energi yang masuk ke sistem (Joule, kJ, dll)
        output_energi (float): Energi keluaran dari sistem (Joule, kJ, dll)
    Return:
        float: Nilai efisiensi (antara 0 - 1)
    """

    if input_energi == 0:
        print("Error: Input energi tidak boleh nol.")
        return None

    efisiensi = output_energi / input_energi

    print("=== Perhitungan Efisiensi Turbin ===")
    print(f"Input Energi   : {input_energi}")
    print(f"Output Energi  : {output_energi}")
    print(f"Efisiensi Turbin: {efisiensi:.2f} atau {efisiensi * 100:.1f}%")

    return efisiensi


hasil = hitung_efisiensi(500, 400)

if hasil is not None:
    print(f"\nNilai efisiensi yang dihitung adalah {hasil:.2f} atau {hasil * 100:.1f}%")

=== Perhitungan Efisiensi Turbin ===
Input Energi   : 500
Output Energi  : 400
Efisiensi Turbin: 0.80 atau 80.0%

Nilai efisiensi yang dihitung adalah 0.80 atau 80.0%


1.Definisi Fungsi: Fungsi hitung_efisiensi didefinisikan, siap untuk digunakan.

2.Panggilan: Fungsi dipanggil dengan input_energi = 500 dan output_energi = 400.

3.Validasi: Sistem memastikan input (500) tidak nol.teks tebal

4.Perhitungan: Efisiensi dihitung: 400 / 500 = 0.8.

5.Cetak Output 1: Rincian perhitungan dicetak: "Efisiensi Turbin: 0.80 atau 80.0%".

6.Kembalian Nilai: Nilai 0.8 dikembalikan dan disimpan di variabel hasil.

7.Cetak Output 2: Program utama mencetak hasil akhir: "Nilai efisiensi yang dihitung adalah 0.80 atau 80.0%".

In [2]:
# SOAL NO 4
def sistem_katup(tekanan_uap, status_manual_override):
    """
    Fungsi ini mensimulasikan logika kerja katup pengaman (safety valve).

    Parameter:
        tekanan_uap (float): nilai tekanan uap (dalam psi)
        status_manual_override (bool): status override manual
                                       True  = katup dikunci manual (tidak boleh terbuka)
                                       False = sistem otomatis aktif (boleh terbuka)

    Return:
        str: status katup ('Terbuka' atau 'Tertutup')
    """

    print("=== SISTEM PEMANTAUAN KATUP PENGAMAN ===")
    print(f"Tekanan Uap Saat Ini     : {tekanan_uap} psi")
    print(f"Status Manual Override   : {status_manual_override}")
    print("-------------------------------------------")

    if tekanan_uap > 150 and status_manual_override == False:
        print("Kondisi terpenuhi:")
        print(" - Tekanan melebihi ambang batas 150 psi")
        print(" - Override manual TIDAK aktif (False)")
        print(">>> Hasil: Katup Terbuka untuk melepaskan tekanan berlebih.")
        status_katup = "Terbuka"
    else:
        print("Kondisi tidak terpenuhi:")
        if tekanan_uap <= 150:
            print(" - Tekanan masih di bawah batas aman (<= 150 psi).")
        if status_manual_override == True:
            print(" - Override manual aktif, sistem dikunci oleh operator.")
        print(">>> Hasil: Katup Tetap Tertutup demi keamanan.")
        status_katup = "Tertutup"

    print("===========================================\n")
    return status_katup



hasil1 = sistem_katup(160, False)

hasil2 = sistem_katup(120, False)

hasil3 = sistem_katup(180, True)

print("=== RANGKUMAN HASIL PENGUJIAN ===")
print(f"Skenario 1 (160 psi, override False): Katup {hasil1}")
print(f"Skenario 2 (120 psi, override False): Katup {hasil2}")
print(f"Skenario 3 (180 psi, override True ): Katup {hasil3}")
print("===================================")

=== SISTEM PEMANTAUAN KATUP PENGAMAN ===
Tekanan Uap Saat Ini     : 160 psi
Status Manual Override   : False
-------------------------------------------
Kondisi terpenuhi:
 - Tekanan melebihi ambang batas 150 psi
 - Override manual TIDAK aktif (False)
>>> Hasil: Katup Terbuka untuk melepaskan tekanan berlebih.

=== SISTEM PEMANTAUAN KATUP PENGAMAN ===
Tekanan Uap Saat Ini     : 120 psi
Status Manual Override   : False
-------------------------------------------
Kondisi tidak terpenuhi:
 - Tekanan masih di bawah batas aman (<= 150 psi).
>>> Hasil: Katup Tetap Tertutup demi keamanan.

=== SISTEM PEMANTAUAN KATUP PENGAMAN ===
Tekanan Uap Saat Ini     : 180 psi
Status Manual Override   : True
-------------------------------------------
Kondisi tidak terpenuhi:
 - Override manual aktif, sistem dikunci oleh operator.
>>> Hasil: Katup Tetap Tertutup demi keamanan.

=== RANGKUMAN HASIL PENGUJIAN ===
Skenario 1 (160 psi, override False): Katup Terbuka
Skenario 2 (120 psi, override False): Katup

Penjelasan Lengkap:

def sistem_katup(...): Fungsi utama untuk mensimulasikan sistem katup pengaman.

Docstring ("""...""") Memberi dokumentasi tentang parameter dan fungsinya, agar mudah dipahami oleh teknisi lain.

Bagian print(...) di awal fungsi Menampilkan data kondisi nyata dari sistem sebelum keputusan dibuat:

Tekanan uap saat ini

Status override manual

if tekanan_uap > 150 and status_manual_override == False: Kondisi majemuk menggunakan operator AND. Kedua kondisi harus benar agar katup boleh terbuka.

Bagian else: Menangani semua kondisi selain itu — misalnya tekanan belum cukup tinggi, atau override aktif.

Percabangan tambahan di dalam else: Memberi tahu alasan spesifik mengapa katup tidak terbuka:

Tekanan belum cukup tinggi

Override manual aktif

Tiga contoh pengujian (test case):

Tekanan tinggi, override nonaktif → katup terbuka

Tekanan rendah, override nonaktif → tetap tertutup

Tekanan tinggi, override aktif → tertutup

In [3]:
# SOAL NO 6
tekanan_uap = 160.0
status_manual_override = False

if tekanan_uap > 150 and status_manual_override == False:
    print("Katup terbuka")
else:
    print("Katup tertutup")


Katup terbuka


Penjelasan singkat:

Operator and digunakan karena kedua kondisi harus terpenuhi.

Kondisi pertama: tekanan_uap > 150

Kondisi kedua: status_manual_override == False

Jika keduanya benar, maka program mencetak "Katup terbuka", selain itu "Katup tertutup".

In [4]:
#SOAL NO 8

def hitung_efisiensi_termal(energi_keluar, energi_masuk):
    efisiensi = (energi_keluar / energi_masuk) * 100
    print(f"Efisiensi termal = {efisiensi:.2f}%")

# Contoh pemanggilan fungsi
hitung_efisiensi_termal(500, 1200)

Efisiensi termal = 41.67%


Penjelasan langkah kerja secara singkat: Soal NO 8: Efisiensi Termal

1.Panggilan Fungsi: Fungsi hitung_efisiensi_termal dipanggil dengan \Energi Keluar = 500 dan Energi Masuk = 1200.

2.Perhitungan: Efisiensi dihitung: (500 / 1200) \times 100 \approx 41.67.

3.Output: Nilai efisiensi dicetak sebagai persentase: Efisiensi termal = 41.67%.

In [8]:
#SOAL NO 10

volume = 0

while volume < 100:
    volume += 10
    if volume == 80:
        print(f"WARNING: Hampir penuh! {volume} liter.")
    elif volume == 100:
        print(f"TANGKI PENUH - Pengisian berhenti. {volume} liter.")
    else:
        print(f"Pengisian: {volume} liter")


Pengisian: 10 liter
Pengisian: 20 liter
Pengisian: 30 liter
Pengisian: 40 liter
Pengisian: 50 liter
Pengisian: 60 liter
Pengisian: 70 liter
Pengisian: 90 liter
TANGKI PENUH - Pengisian berhenti. 100 liter.


Penjelasan langkah kerja singkat Soal NO 10: Simulasi Pengisian

1.Inisialisasi: volume dimulai dari 0.

2.Looping (10 L s/d 70 L): Loop while berjalan 7 kali. volume bertambah 10 dan mencetak status "Pengisian: X liter" (7 output).

3.Peringatan (80 L): volume mencapai 80. Kondisi if volume == 80 terpenuhi. Output: "WARNING: Hampir penuh! saat 80 liter".

4.Lanjutan (90 L): volume bertambah menjadi 90. Kondisi else terpenuhi. Output: "Pengisian: 90 liter".

5.Penuh (100 L): volume bertambah menjadi 100. Kondisi elif volume == 100 terpenuhi. Output: "TANGKI PENUH - Pengisian berhenti. saat 100 liter".

6.Selesai: Loop berhenti karena 100.